src/tests/ftest/osa/osa_offline_parallel_test.py

```diff
@@ -0,0 +1,229 @@
+                dmg.pool_drain(puuid, rank)
+            elif action == "reintegrate":
+                time.sleep(30)
+                dmg.pool_reintegrate(puuid, (rank + 1), target)
+            else:
+                self.fail("Invalid action for dmg thread")
+        except CommandFailure as _error:
+            results.put("{} failed".format(action))
+            # elif action == "extend":
+            #    dmg.pool_extend(puuid, (rank + 2))
+
+    def run_offline_parallel_test(self, num_pool, data=False):
+        """Run the offline reintegration without data.
+            Args:
+            num_pool (int) : total pools to create for testing purposes.
+            data (bool) : whether pool has no data or to create
+                          some data in pool. Defaults to False.
+        """
+        # Create a pool
+        pool = {}
+        pool_uuid = []
+        target_list = []
+
+        # Exclude target : random two targets (target idx : 0-7)
+        n = random.randint(0, 6)
+        target_list.append(n)
+        target_list.append(n+1)
+        t_string = "{},{}".format(target_list[0], target_list[1])
+
+        # Exclude rank 2.
+        rank = 2
+
+        for val in range(0, num_pool):
+            pool[val] = TestPool(self.context,
+                                 dmg_command=self.get_dmg_command())
+            pool[val].get_params(self)
+            # Split total SCM and NVME size for creating multiple pools.
+            pool[val].scm_size.value = int(pool[val].scm_size.value /
+                                           num_pool)
+            pool[val].nvme_size.value = int(pool[val].nvme_size.value /
+                                            num_pool)
+            pool[val].create()
+            pool_uuid.append(pool[val].uuid)
+            self.pool = pool[val]
+            if data:
+                self.write_single_object()
+
+        # Exclude and reintegrate the pool_uuid, rank and targets
+        for val in range(0, num_pool):
+            self.pool = pool[val]
+            self.pool.display_pool_daos_space("Pool space: Beginning")
+            pver_begin = self.get_pool_version()
+            self.log.info("Pool Version at the beginning %s", pver_begin)
+            # Create the threads here
+            threads = []
+            osa_tasks = ["drain", "exclude", "reintegrate"]
+            i = 0
+            for action in osa_tasks:
+                # Add a dmg thread
+                process = threading.Thread(target=self.dmg_thread,
+                                           kwargs={"puuid":
+                                                   self.pool.uuid,
+                                                   "rank": rank,
+                                                   "target": t_string,
+                                                   "action": action,
+                                                   "results":
+                                                   self.out_queue})
```

Comment:
(pylint-no-member) Instance of 'OSAOfflineParallelTest' has no 'out_queue' member